File: VB/11_SecurityAndSignatures/SignedByLTV.vb
Imports Spire.Pdf
Imports Spire.Pdf.Security
Imports System.Security.Cryptography.X509Certificates
Imports Spire.Pdf.Graphics

Namespace SignedByLTV
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Load the input file path
			Dim inputFile As String = "..\..\..\..\..\..\Data\DigitalSignature.pdf"

			' Create a new PdfDocument object
			Dim doc As New PdfDocument()

			' Load the PDF document from the input file path
			doc.LoadFromFile(inputFile)

			' Get the first page of the document
			Dim page As PdfPageBase = doc.Pages(0)

			' Set the path of the certificate .pfx file
			Dim pfxPath As String = "..\..\..\..\..\..\Data\gary.pfx"

			' Load the certificate from the pfx file with the specified password and exportable flag
			Dim cer As New PdfCertificate(pfxPath, "e-iceblue", X509KeyStorageFlags.Exportable)

			' Add a signature to the specified position on the page
			Dim signature As New PdfSignature(doc, page, cer, "signature")
			signature.Bounds = New RectangleF(New PointF(90, 550), New SizeF(180, 90))

			' Set the content of the signature
			signature.NameLabel = "Digitally signed by: Gary"
			signature.LocationInfoLabel = "Location:"
			signature.LocationInfo = "CN"
			signature.ReasonLabel = "Reason: "
			signature.Reason = "Ensure authenticity"
			signature.ContactInfoLabel = "Contact Number: "
			signature.ContactInfo = "028-81705109"
			signature.DocumentPermissions = PdfCertificationFlags.AllowFormFill Or PdfCertificationFlags.ForbidChanges
			signature.GraphicsMode = GraphicMode.SignImageAndSignDetail
			signature.SignImageSource = PdfImage.FromFile("..\..\..\..\..\..\Data\logo.png")

			' Configure OCSP (Online Certificate Status Protocol) for the signature
			signature.ConfigureHttpOCSP(Nothing, Nothing)

			' Save the PDF document with the signature to the output file path
			Dim outputFile As String = "result.pdf"
			doc.SaveToFile(outputFile, FileFormat.PDF)

			' Close the PDF document
			doc.Close()

			'Launch the file
			PDFDocumentViewer(outputFile)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/11_SecurityAndSignatures/SignedByLTV.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Security;
using System.Security.Cryptography.X509Certificates;
using Spire.Pdf.Graphics;

namespace SignedByLTV
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            string inputFile = @"..\..\..\..\..\..\Data\DigitalSignature.pdf";

            // Load a PDF document
            PdfDocument doc = new PdfDocument();
            doc.LoadFromFile(inputFile);

            // Get the first page
            PdfPageBase page = doc.Pages[0];

            // Load a certificate .pfx file
            String pfxPath = @"..\..\..\..\..\..\Data\gary.pfx";
            PdfCertificate cer = new PdfCertificate(pfxPath, "e-iceblue", X509KeyStorageFlags.Exportable);

            // Add a signature to the specified position
            PdfSignature signature = new PdfSignature(doc, page, cer, "signature");
            signature.Bounds = new RectangleF(new PointF(90, 550), new SizeF(180, 90));

            // Set the signature content
            signature.NameLabel = "Digitally signed by: Gary";
            signature.LocationInfoLabel = "Location:";
            signature.LocationInfo = "CN";
            signature.ReasonLabel = "Reason: ";
            signature.Reason = "Ensure authenticity";
            signature.ContactInfoLabel = "Contact Number: ";
            signature.ContactInfo = "028-81705109";
            signature.DocumentPermissions = PdfCertificationFlags.AllowFormFill | PdfCertificationFlags.ForbidChanges;
            signature.GraphicsMode = GraphicMode.SignImageAndSignDetail;
            signature.SignImageSource = PdfImage.FromFile(@"..\..\..\..\..\..\Data\logo.png");

            // Configure OCSP (Online Certificate Status Protocol)
            signature.ConfigureHttpOCSP(null, null);

            // Save the PDF file
            string outputFile = "result.pdf";
            doc.SaveToFile(outputFile, FileFormat.PDF);

            //Launch the file
            PDFDocumentViewer(outputFile); 
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
